torch/csrc/api/src/nn/modules/transformer.cpp

```diff
@@ -72,7 +72,8 @@ Tensor TransformerEncoderLayerImpl::forward(
     src2 = linear2(dropout(F::relu(linear1(ret))));
   }
   else {
-    TORCH_CHECK(false, "activation should be kGELU/kReLU, not ", torch::enumtype::get_enum_name(options.activation()));
+    auto callable_activation = *c10::get_if<std::function<Tensor(const Tensor&)> >(&options.activation());
```

Comment:
The earlier code implies that it was possible previously to find something here besides kGELU/kRELU... if that's true, then wouldn't this new code open be vulnerable to segfaulting? OTOH maybe you realized the earlier dynamic check was superfluous - from first templating principles it seems like it would be...

Reply:
Hmm, great question- I think it's probably best to be safe and add the final `else` branch back in. I think there's always the possibility of some weird casting trickery that could cause that branch to be triggered. It also helps guard against other variant values being added in the future.